Add basic_logger utility

taldcroft opened the pull request:
## Description

This implements a thin wrapper around `logging.basicConfig`, addressing the show-stopper that `basicConfig` only configures the root logger.

I copied the `basicConfig` docs and then customized/enhanced them a bit.

@javierggt - what do you think? It's a bit of a hack but maybe works for us?

## Interface impacts
<!-- API changes, file format updates, coordination of changes with the community. -->
None

## Testing
<!-- If relevant describe any special setup for testing. -->

### Unit tests
<!-- At least one of these must be checked if unit tests exist. DELETE the unchecked/untested options. -->
- [x] No unit tests

### Functional tests
<!-- Describe and document results of any functional tests, otherwise leave the text below -->
```
In [1]: from ska_helpers.logging import basic_logger

In [2]: logger = basic_logger('me')

In [3]: logger.info('asdf')

In [5]: logger.warning('asdf')
2022-06-09 07:15:47,349 <module>: asdf

In [6]: logger = basic_logger('me', level='INFO')

In [7]: logger.info('asdf')

In [8]: logger = basic_logger('me', level='INFO', force=True)

In [9]: logger.info('asdf')
2022-06-09 07:16:25,243 <module>: asdf

In [10]: logger = basic_logger('me', level='INFO', force=True, format=None)

In [11]: logger.info('asdf')
INFO:me:asdf
```

javierggt reviewed (changes requested):
I think it is ok. I checked that it works as advertised.

I would note that this does not help in the more complicated logging cases, but it is good as a replacement of pyyaks with what I think is a more predictable behavior.

My first thought was "what about something else changes in the global scope of the logging module while it is monkey-patched?". I checked, and there is not much else in the global scope. It is still something to keep in mind in the future.

My only quibble is that perhaps it should go into a new sub-module `ska_helpers.logging`. Otherwise this `utils` thing will eventually become a mix-match of tools and functions. It is also possible that we come up with other logging tools. For example, aca_view has a `logging` module with some logging utils that we could move into ska_helpers if we ever wanted to use them elsewhere (https://github.com/sot/aca_view/blob/master/aca_view/logging.py).

taldcroft commented:
@javierggt - I implemented your suggestion and added something to the docstring about possible multi-thread/process issues.

I re-did the functional testing shown in the description and got the same results.
